app/_src/gateway/upgrade-31x.md

```diff
@@ -45,12 +45,21 @@ The following table outlines various upgrade path scenarios to 3.1.x depending o
 
 | **Current version** | **Topology** | **Direct upgrade possible?** | **Upgrade path** |
 | ------------------- | ------------ | ---------------------------- | ---------------- |
-| 2.x | Traditional | No | [Upgrade to 2.8.2.x](/gateway/2.8.x/install-and-run/upgrade-enterprise/) (required for blue/green deployments only), then [upgrade to 3.0.x](/gateway/3.0.x/upgrade/), and then [upgrade to 3.1.x](#upgrade-from-30x-to-31x). |
-| 2.x | Hybrid | No | [Upgrade to 2.8.2.x](/gateway/2.8.x/install-and-run/upgrade-enterprise/), then [upgrade to 3.0.x](/gateway/3.0.x/upgrade/), and then [upgrade to 3.1.x](#upgrade-from-30x-to-31x). |
-| 2.x | DB less | No | [Upgrade to 3.0.x](/gateway/3.0.x/upgrade/), and then [upgrade to 3.1.x](#upgrade-from-30x-to-31x). |
-| 3.0.x | Traditional | Yes | [Upgrade to 3.1.x](#upgrade-from-30x-to-31x). |
-| 3.0.x | Hybrid | Yes | [Upgrade to 3.1.x](#upgrade-from-30x-to-31x). |
-| 3.0.x | DB less | Yes | [Upgrade to 3.1.x](#upgrade-from-30x-to-31x). |
+| 2.x–2.7.2.0 | Traditional | No | [Upgrade to 2.8.2.x](/gateway/2.8.x/install-and-run/upgrade-enterprise/) (required for blue/green deployments only), then [upgrade to 3.0.x](/gateway/3.0.x/upgrade/), and then [upgrade to 3.1.x](#migrate-db). |
+| 2.x–2.7.2.0 | Hybrid | No | [Upgrade to 2.8.2.x](/gateway/2.8.x/install-and-run/upgrade-enterprise/), then [upgrade to 3.0.x](/gateway/3.0.x/upgrade/), and then [upgrade to 3.1.x](#migrate-db). |
+| 2.x–2.7.2.0 | DB less | No | [Upgrade to 3.0.x](/gateway/3.0.x/upgrade/), and then [upgrade to 3.1.x](#migrate-db). |
+| 2.8.x.x | Traditional | Only if you upgrade to 3.1.1.3 | [Upgrade to 3.1.1.3](#migrate-db). |
+| 2.8.x.x | Hybrid | Only if you upgrade to 3.1.1.3 | [Upgrade to 3.1.1.3](#migrate-db). |
+| 2.8.x.x | DB less | Only if you upgrade to 3.1.1.3 | [Upgrade to 3.1.1.3](#migrate-db). |
+| 3.0.x | Traditional | Yes | [Upgrade to 3.1.1.3](#migrate-db). |
+| 3.0.x | Hybrid | Yes | [Upgrade to 3.1.1.3](#migrate-db). |
+| 3.0.x | DB less | Yes | [Upgrade to 3.1.1.3](#migrate-db). |
+
+{% if_version gte: 3.2.x %}
+| 3.1.x | Traditional | Yes | [Upgrade to 3.2.x](#migrate-db). |
+| 3.1.x | Hybrid | Yes | [Upgrade to 3.2.x](#migrate-db). |
+| 3.1.x | DB less | Yes | [Upgrade to 3.2.x](#migrate-db). |
+{% endif_version %}
```

Comment:
For traditional and dbless, any 3.0.x and 3.1.x can upgrade directly to 3.2.x.
For hybrid, 3.1.x _must_ upgrade to 3.1.1.3 before upgrading to 3.2.x.
I _believe_ that 3.0.x hybrid can upgrade directly to 3.2.x as well, but I need to double check. Stand by...

```suggestion
| 3.0.x | Traditional | Yes | [Upgrade to 3.2.x](#migrate-db). |
| 3.0.x | Hybrid | Yes | [Upgrade to 3.2.x](#migrate-db). |
| 3.0.x | DB less | Yes | [Upgrade to 3.2.x](#migrate-db). |

{% if_version gte: 3.2.x %}
| 3.1.x | Traditional | Yes | [Upgrade to 3.2.x](#migrate-db). |
| 3.1.x | Hybrid | Yes | [Upgrade to 3.1.1.3](#migrate-db). |
| 3.1.x | DB less | Yes | [Upgrade to 3.2.x](#migrate-db). |
{% endif_version %}
```